data/src/error_analysis/erroranalysis.py

```diff
@@ -0,0 +1,77 @@
+"""Error analysis tool for G2P.
+
+This script assumes 2 input files. a) covering grammar b) test output.
+CG file: Each line contain grapheme and their corresponding pronunciation
+separated by a tab
+Test file: contains three attributes in each line separated by a tab.
+Example: espresso ɛ s p ɹ ɛ s ə ʊ     ɛ k s p ɹ ɛ s ə ʊ
+ """
+
+__author__ = "Arundhati Sengupta"
+
+import pynini
+import argparse
+from prettytable import PrettyTable
+
+
+def main(args: argparse.Namespace) -> None:
+    cg_fst = pynini.string_file(args.cg_path).closure().optimize()
+    rulematch_predmatch = 0
+    rulematch_pred_notmatch = 0
+    not_rulematch_predmatch = 0
+    not_rulematch_pred_notmatch = 0
+    total_records = 0
+    with open(args.test_path, "r") as source:
+        for line in source:
+            total_records += 1
+            parts = line.split("\t")
+            lg = parts[0].strip()
+            act = parts[1].replace(" ", "").replace(".", "").strip()
+            pred_pron = parts[2].replace(" ", "").replace(".", "").strip()
+            lattice = (lg @ cg_fst @ pred_pron).project(True)
+            if lattice.start() == pynini.NO_STATE_ID:
+                if act == pred_pron:
+                    not_rulematch_predmatch += 1
+                else:
+                    not_rulematch_pred_notmatch += 1
+            else:
+                if act == pred_pron:
+                    rulematch_predmatch += 1
+                else:
+                    rulematch_pred_notmatch += 1
+
+    print("Total Number of Records", total_records)
+    rule_m_pred_nm = (rulematch_pred_notmatch / total_records) * 100
```

Comment:
Easier way to write this (and following lines):

```
rule_m_pred_nm = 100 * rulematch_pred_notmatch / total_records
```

I like this because parentheses make things harder to read.

Reply:
fixed.